src/constraint_framework/4C_constraint_framework_submodelevaluator_nullspace.cpp

```diff
@@ -0,0 +1,178 @@
+// This file is part of 4C multiphysics licensed under the
+// GNU Lesser General Public License v3.0 or later.
+//
+// See the LICENSE.md file in the top-level for license information.
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include "4C_constraint_framework_submodelevaluator_nullspace.hpp"
+
+#include "4C_fem_condition.hpp"
+#include "4C_fem_discretization_nullspace.hpp"
+#include "4C_fem_general_elementtype.hpp"
+#include "4C_io_pstream.hpp"
+#include "4C_linalg_utils_sparse_algebra_math.hpp"
+#include "4C_linalg_utils_sparse_algebra_print.hpp"
+#include "4C_structure_new_timint_base.hpp"
+#include "4C_structure_new_timint_basedataglobalstate.hpp"
+
+FOUR_C_NAMESPACE_OPEN
+
+/*----------------------------------------------------------------------------*
+ *----------------------------------------------------------------------------*/
+Constraints::SubmodelEvaluator::NullspaceConstraintManager::NullspaceConstraintManager(
+    std::shared_ptr<Core::FE::Discretization> discret_ptr)
+{
+  discret_ptr_ = discret_ptr;
+
+  std::vector<const Core::Conditions::Condition*> KSPcond;
+  discret_ptr_->get_condition("KrylovSpaceProjection", KSPcond);
+  int numcond = KSPcond.size();
+  int numsolid = 0;
+
+  const Core::Conditions::Condition* kspcond = nullptr;
+  // check if for solid Krylov projection is required
+  for (int icond = 0; icond < numcond; icond++)
+  {
+    const std::string& name = KSPcond[icond]->parameters().get<std::string>("DIS");
+    if (name == "structure")
+    {
+      numsolid++;
+      kspcond = KSPcond[icond];
+    }
+  }
+
+  if (numsolid != 1)
+  {
+    FOUR_C_THROW(
+        "For the nullspace constraint enforcement exactly one KrylovSpaceCondition for "
+        "the structure field is necessary.");
+  }
+
+  const auto type = kspcond->parameters().get<std::string>("TYPE");
+
+  if (type != "constraint")
+  {
+    FOUR_C_THROW(
+        "The KrylovSpaceCondition needs to be of type constraint to be properly enforced.");
+  }
+
+  nullspace_dimension_ = kspcond->parameters().get<int>("NUMMODES");
+
+  // just grab the block information on the first element that appears and check with given number
+  Core::Elements::Element* dwele = discret_ptr->l_row_element(0);
+  int nullspace_dimension;
+  dwele->element_type().nodal_block_information(dwele, number_of_dofs_, nullspace_dimension);
```

Comment:
This encodes the assumption, that all elements in the discretization are of the same type. Should we check if that is really the case (in release or only in debug mode)? Should we warn the user about this assumption?